kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
+                         'credentials to the environment. Please see\n ' +
+                         'https://github.com/Kaggle/kaggle-api#api-credentials '
+                          + 'for instructions.')
 
-    def set_config_value(self, name, value, quiet=False):
+
+    def _authenticate_config(self, quiet=False):
+        '''autheticate config is the first effort to get a username
+           and key to authenticate to the Kaggle API. Since we can get the
+           username and password from the environment, it's not required.
+
+           Parameters
+           ==========
+           configuration: the Configuration object to save a username and
+                          password, if defined
+           quiet: add verbosity
+        '''
+        config_data = {}
+
+        if os.path.exists(self.config):
+
+            try:
+                if os.name != 'nt':
+                    permissions = os.stat(self.config).st_mode
+                    if (permissions & 4) or (permissions & 32):
+                        print('Warning: Your Kaggle API key is readable by other' 
+                              'users on this system! To fix this, you can run' + 
+                              '\'chmod 600 {}\''.format(self.config))
+
+                with open(self.config, 'r') as f:
+                    config_data = json.load(f)
+            except:
+                pass
+
+        else:
+
+            # Warn the user that configuration will be reliant on environment
+            if not quiet:
+                print('No Kaggle API config file found, will use environment.')
+
+        return config_data
+
+  
+    def _read_config(self):
+        '''read in the configuration file, a json file defined at self.confi'''
+        
         try:
             with open(self.config, 'r') as f:
                 config_data = json.load(f)
-            if value is not None:
-                config_data[name] = value
-                with open(self.config, 'w') as f:
-                    json.dump(config_data, f, indent=2)
-            if name in config_data:
-                self.config_values[name] = config_data[name]
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        except FileNotFoundError:
+            config_data = {}
+
+        return config_data
+
+
+    def _write_config(self, config_data, indent=1):
+        '''write config data to file.
+        '''
+        with open(self.config, 'w') as f:
+            json.dump(config_data, f, indent=indent)
+
+
+    def set_config_value(self, name, value, quiet=False):
+        '''a client helper function to set a configuration value, meaning
+           reading in the configuration file (if it exists), saving a new
+           config value, and then writing back
+        '''
+
+        config_data = self._read_config()
+   
+        # If defined by client, set and save!
+        self._write_config(config_data)
```

Comment:
Doesn't this need to be within the below `if` block?

Reply:
Yes, I'm not sure why I left it there.